ahachul_backend/src/main/kotlin/backend/team/ahachul_backend/common/client/AwsS3Client.kt

```diff
@@ -0,0 +1,40 @@
+package backend.team.ahachul_backend.common.client
+
+import backend.team.ahachul_backend.common.properties.AwsS3Properties
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.PutObjectRequest
+import org.springframework.stereotype.Component
+import org.springframework.web.multipart.MultipartFile
+import java.io.File
+import java.io.FileOutputStream
+import java.util.UUID
+
+@Component
+class AwsS3Client(
+    private val s3Client: AmazonS3Client,
+    private val s3Properties: AwsS3Properties
+) {
+
+    fun upload(file: MultipartFile) {
+        s3Client.putObject(
+            PutObjectRequest(
+                s3Properties.bucketName,
+                generateRandomUUID(),
+                convertToFile(file)
+            )
+        )
+    }
+
+    private fun generateRandomUUID(): String {
+        return UUID.randomUUID().toString()
+    }
+
+    private fun convertToFile(multipartFile: MultipartFile): File {
+        val file = File(multipartFile.originalFilename!!)
+        file.createNewFile()
+        val fos = FileOutputStream(file)
+        fos.write(multipartFile.bytes)
```

Comment:
이렇게 File을 만들어서 보내면 무거운 파일 쓰기 작업 + 불필요하게 로컬에 파일이 저장되지 않나요?

해당 함수 사용하시면 `inputStream` 으로 바이트만 전달하고 나머지 파일에 대한 정보는 `ObjectMetadata` 로 보낼 수 있습니다! 그리고 예외 처리도 해주는게 좋을 것 같아요

<img width="456" alt="image" src="https://user-images.githubusercontent.com/71436576/236712680-a9fedfde-a1d9-4d01-ad31-f1ed74b0c01a.png">

```suggestion
        try (inputStream: InputStream = file.getInputStream()) {
               S3Client.putObject(
                        PutObjectRequest(bucket, fileName, inputStream, objectMetaData)
         } catch(e: IOException) {
             // 로그 남기고 런타임 예외로 변환해서 던지기
         }
```

Reply:
아 코틀린에서는 [use 확장 함수](https://www.baeldung.com/kotlin/try-with-resources)로  try-with-resouces를 대체하는군요..! 자바랑 헷갈렷네요

```suggestion
        try {
            inputStream: InputStream = file.getInputStream()
            inputStream.use {
               S3Client.putObject(...)
            }
        } catch (e: IOException) {
             // 런타임 예외 변환
        }
```